src/gatherers/user-timings.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Gather = require('./gather');
+const TimelineModel = require('../lib/traces/devtools-timeline-model');
+
+class UserTimings extends Gather {
+  get name() {
+    return 'userTimings';
+  }
+
+  // Extract blink.user_timing events generated by the site
+  postProfiling(options, tracingData) {
+    if (typeof tracingData === 'undefined' ||
+        !tracingData.traceContents ||
+        !Array.isArray(tracingData.traceContents)) {
+      this.artifact = [];
+      return;
+    }
+
+    let userTimings = [];
+    let navigationStartTime;
+    let measuresStartTimes = {};
+    let traceStart = false;
+
+    // Fetch blink.user_timing events from the tracing data
+    const timelineModel = new TimelineModel(tracingData.traceContents);
+    const modeledTraceData = timelineModel.timelineModel();
+    modeledTraceData.mainThreadEvents().filter(
+      // Get all blink.user_timing events
+      ut => ut.hasCategory('blink.user_timing') || ut.name === 'TracingStartedInPage'
+    ).forEach(ut => {
+      if (ut.phase === 'R' || ut.phase === 'I') {
+        // Mark event
+        if (ut.name === 'TracingStartedInPage' && !traceStart) {
```

Comment:
is it possible to have multiple `TracingStartedInPage`s? If so, is the first one always the right one to use?